addons/udes_stock_refactoring/models/stock_picking.py

```diff
@@ -0,0 +1,367 @@
+
+            @param (optional) origin
+                Search for stock.picking records based on the origin
+                field. Needs to be a complete match.
+
+            @param (optional) package_name
+                Search of stock.pickings associated with a specific
+                package_name (exact match).
+
+            @param (optional) product_id
+                If it is set then location_id must also be set and stock.pickings
+                are found using both of those values (states is optional).
+
+            @param (optional) location_id
+                If it is set then only internal transfers acting on that
+                location are considered. In all cases, if states is set
+                then only pickings in those states are considered.
+
+            @param (optional) backorder_id
+                Id of the backorder picking. If present, pickings are found
+                by backorder_id and states.
+
+            (IGNORE FOR NOW) @param (optional) allops: Boolean.
+                If True, all pack operations are included. If False, only
+                pack operations that are for the pallet identified by param
+                pallet (and it's sub-packages) are included.
+                Defaults to True.
+
+            @param (optional) states
+                A List of strings that are states for pickings. If present
+                only pickings in the states present in the list are
+                returned.
+                Defaults to all, possible values:
+                'draft', 'cancel', 'waiting', 'confirmed', 'assigned', 'done'
+
+            @param (optional) result_package_id
+                If an id is supplied all pickings that are registered to
+                this package id will be returned. This can also be used
+                in conjunction with the states parameter
+
+            @param (optional) picking_priorities
+                When supplied all pickings of set priorities and states
+                will be searched and returned
+
+            @param (optional) picking_ids
+                When supplied pickings of the supplied picking ids will
+                be searched and returned. If used in conjunction with
+                priorities then only those pickings of those ids will be
+                returned.
+
+            @param (optional) picking_type_ids: Array (int)
+                If it is set the pickings returned will be only from the picking types in the array.
+        """
+        Picking = self.env["stock.picking"]
+        Package = self.env["stock.quant.package"]
+        Users = self.env["res.users"]
+
+        order = None
+
+        if states is None:
+            states = ["draft", "cancel", "waiting", "confirmed", "assigned", "done"]
+
+        warehouse = Users.get_user_warehouse()
+        if picking_type_ids is None:
+            picking_type_ids = warehouse.get_picking_types().ids
+
+        if self:
```

Comment:
Lots of ifs statements here, that look to just overwrite the domain that may have previously been set if someone is added lots of kwargs. If they should only set one, or a subset can we create a mapping to shrink it?

Reply:
This giant function is no longer needed and has now been removed. :)